src/Sulu/Bundle/AdminBundle/Resources/js/containers/ViewRenderer/ViewRenderer.js

```diff
@@ -43,17 +44,24 @@ export default class ViewRenderer extends React.Component<Props> {
             throw new Error('View "' + view + '" has not been found');
         }
 
+        // if any view has sidebar - the sidebar will be visible
+        hasSidebar = View.hasSidebar ? true : hasSidebar;
+
         const element = (
             <View router={router} route={route} key={this.getKey(route)}>
                 {(props) => child ? React.cloneElement(child, props) : null}
             </View>
         );
 
         if (!route.parent) {
+            if (!hasSidebar) {
+                sidebarStore.clearConfig();
```

Comment:
Shouldn't something like this happen always anyway? Resp. should this be done somewhere else, in case the sidebar does not return a config for the sidebar?

Reply:
Exactly this line seems to cause the following error now:

```
Warning: Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.
```

Although it says we should `componentWillMount` instead, I would not advise to do so, because this method is already deprecated. It should probably be moved to `componentDidUpdate` as suggested in their [upgrade blog post](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html).

Would you have time to look into that?